src/main/java/io/github/genomicdatainfrastructure/daam/services/SubmitApplicationService.java

```diff
@@ -5,38 +5,65 @@
 package io.github.genomicdatainfrastructure.daam.services;
 
 import static io.github.genomicdatainfrastructure.daam.security.PostAuthenticationFilter.USER_ID_CLAIM;
+
+import io.github.genomicdatainfrastructure.daam.exceptions.ApplicationNotFoundException;
+import io.github.genomicdatainfrastructure.daam.exceptions.ApplicationNotInCorrectStateException;
+import io.github.genomicdatainfrastructure.daam.exceptions.UserNotApplicantException;
 import io.github.genomicdatainfrastructure.daam.remote.rems.api.RemsApplicationCommandApi;
+import io.github.genomicdatainfrastructure.daam.remote.rems.api.RemsApplicationsApi;
 import io.github.genomicdatainfrastructure.daam.remote.rems.model.SubmitApplicationCommand;
 import io.quarkus.oidc.runtime.OidcJwtCallerPrincipal;
 import io.quarkus.security.identity.SecurityIdentity;
 import jakarta.enterprise.context.ApplicationScoped;
 import jakarta.inject.Inject;
 import org.eclipse.microprofile.config.inject.ConfigProperty;
 import org.eclipse.microprofile.rest.client.inject.RestClient;
-
+import jakarta.ws.rs.WebApplicationException;
 
 @ApplicationScoped
 public class SubmitApplicationService {
 
     private final SecurityIdentity identity;
     private final String remsApiKey;
     private final RemsApplicationCommandApi remsApplicationCommandApi;
+    private final RemsApplicationsApi remsApplicationsApi;
 
     @Inject
     public SubmitApplicationService(
         @ConfigProperty(name = "quarkus.rest-client.rems_yaml.api-key") String remsApiKey,
         SecurityIdentity identity,
-        @RestClient RemsApplicationCommandApi applicationsApi
+        @RestClient RemsApplicationCommandApi applicationCommandApi,
+        @RestClient RemsApplicationsApi applicationsApi
     ) {
         this.remsApiKey = remsApiKey;
         this.identity = identity;
-        this.remsApplicationCommandApi = applicationsApi;
+        this.remsApplicationCommandApi = applicationCommandApi;
+        this.remsApplicationsApi = applicationsApi;
     }
 
     public void submitApplication(String id) {
         var principal = (OidcJwtCallerPrincipal) identity.getPrincipal();
         String userId = principal.getClaim(USER_ID_CLAIM);
 
+        try {
```

Comment:
suggestion: to improve readability, you should move the try/catch that captures `40x` to another function, that retrieves the application if no `40x` happens.